parsers/UY.py

```diff
@@ -46,118 +62,182 @@ def get_salto_grande(session: Optional[Session]) -> float:
+        datum["solar"] = max(datum["solar"], 0)
+
+        # ingest date field
+        datefield = hour.find("hora").contents[0]
+        datestr = re.findall("\d\d/\d\d/\d\d\d\d \d+:\d\d", str(datefield))[0]
+        date = arrow.get(datestr, "DD/MM/YYYY h:mm").replace(
+            tzinfo=dateutil.tz.gettz(tz)
+        )
+        datum["time"] = date
+
+        # add to list
+        hour_recs.append(datum)
+    return hour_recs
+
+
+def get_entry_list(
+    session: Session, make_output: Callable[[dict], dict], correct_hydro=True
+) -> dict:
+    """
+    Creates list of return datapoints given make_output function
 
-    datefield = soup.find(
-        "span", attrs={"id": "ctl00_ContentPlaceHolder1_lblUltFecScada"}
-    )
-    datestr = re.findall("\d\d/\d\d/\d\d\d\d \d+:\d\d", str(datefield.contents[0]))[0]
-    date = arrow.get(datestr, "DD/MM/YYYY h:mm").replace(tzinfo=dateutil.tz.gettz(tz))
-
-    table = soup.find(
-        "table", attrs={"id": "ctl00_ContentPlaceHolder1_gridPotenciasNivel1"}
-    )
-
-    obj = {"datetime": date.datetime}
-
-    for tr in table.find_all("tr"):
-        tds = tr.find_all("td")
-        if not len(tds):
-            continue
-
-        key = tds[0].find_all("b")
-        # Go back one level up if the b tag is not there
-        if not len(key):
-            key = tds[0].find_all("font")
-        k = key[0].contents[0]
-
-        value = tds[1].find_all("b")
-        # Go back one level up if the b tag is not there
-        if not len(value):
-            value = tds[1].find_all("font")
-        v_str = value[0].contents[0]
-        if v_str.find(",") > -1 and v_str.find(".") > -1:
-            # there can be values like "1.012,5"
-            v_str = v_str.replace(".", "")
-            v_str = v_str.replace(",", ".")
-        else:
-            # just replace decimal separator, like "125,2"
-            v_str = v_str.replace(",", ".")
-        v = float(v_str)
-
-        # solar reports -0.1 at night, make it at least 0
-        v = max(v, 0)
-
-        obj[k] = v
+    The function will first fetch data for all avaliable times,
+    then use make_output to format the data points before returning
+    a 24 hr history of data entries
+    """
+    # handle datetime here, so we only have to do it once
+    # get data from webpage
+    raw_entries = parse_page(session)
 
-    # https://github.com/tmrowco/electricitymap/issues/1325#issuecomment-380453296
-    salto_grande = get_salto_grande(session)
-    obj["Hidráulica"] = obj.get("Hidráulica", 0.0) + salto_grande
+    entries = []
+    for raw_entry in raw_entries:
+        if correct_hydro:
+            # https://github.com/tmrowco/electricitymap/issues/1325#issuecomment-380453296
+            raw_entry = correct_for_salto_grande(raw_entry, session)
+        entries.append(make_output(raw_entry))
 
-    return obj
+    return entries
+
+
+def fetch_consumption(
+    zone_key: str = "UY",
+    session: Optional[Session] = Session(),
```

Comment:
I think this one is on me but session should not be optional.

Reply:
should this also be the case for `fetch_production`? I'm confused cause this optional Session seems to be in line with the format defined in the wiki.

Reply:
Technically it is in like with the Wiki but we will be moving away from it, I just haven't had the time to update the examples and wiki yet. The new method makes it more clear and generate less code and type errors.

Reply:
So, when I update that the parser starts failing the interface test on CI. Is this okay, or is there some way I should fix that error